### ISSUE TITLE
Should `webargs.fields.Nested` be renamed/aliased to avoid confusion with `marshmallow.fields.Nested`?

### ISSUE THREAD

**Issue by sirosen** [marshmallow-code/webargs#677]
## Context

`webargs.fields` provides all of the names from `marshmallow.fields`, but it rewrites `Nested`:
https://github.com/marshmallow-code/webargs/blob/0b23a8dd3bf464ca447d759fdbdc373f97666fb7/src/webargs/fields.py#L26

This is the only field which changes behavior between the two `fields` modules, since there are no other redefinitions. It's possible that `marshmallow` introduces something named `DelimitedList` or other conflicting names in the future, but (1) that's not very likely to happen in practice (2) we could deal with it when it happens. So, focusing on `Nested`...

#671 is an instance of type confusion surrounding this. I'm not yet sure if it's user error or if we've confused a type checker.

## Proposed Resolution

Rename `Nested` (gradually) in `webargs.fields`. Possible names: `WebargsNested`, `WNested`, `NestedArg`, `DictNested` (because it accepts a dict).

This can be done by renaming the class and assigning, e.g. `Nested = WebargsNested`.
A deprecation warning can be added later for the old name when passed a dict, prior to a major version which drops it.

## (rejected) Alternative Solutions

I'm happy to be convinced that one of these is better than my proposal. But these are what I considered.

### Do nothing

Always an option. But I look at that ignored redefinition type error as a warning sign that we could trip over this in the future.
Plus, it is confusing that `Nested` doesn't quite match the `marshmallow` docs.

### Give `marshmallow.fields.Nested` the ability to accept a dict

Obsoletes `webargs.fields.Nested`.

This is very simple, and if dict-based definition is useful to `webargs` users, maybe it's useful to all `marshmallow` users.
However, I'd only pursue this with buy-in from the `marshmallow` maintainers.

### Drop support for dict-based definition in `webargs.fields.Nested`

Basically, remove the custom subclass. It's obviously the "best" for maintenance.

However, it's probable that users are using and like this feature. When we considered removing all built-in support for dict definition in the past, users informed us that they find it useful. The utility of dict definition _might_ not hold for `Nested` in the same way as simple query param parsing, but I have no evidence for that.

**Comment by lafrech** [comment]
> Give marshmallow.fields.Nested the ability to accept a dict

This makes sense to me.

(This reminded me of https://github.com/marshmallow-code/marshmallow/issues/1759 but it is not the same thing.)

Your proposal in #681 doesn't make the code much more complex, but it makes it more complicated to grasp for a user. Solving this in marshmallow by accepting dicts in `ma.fields.Nested` means the user can ignore all of this stuff. No surprise.

I had a quick look at `Nested` code and I think it would fit without making things awkwardly complicated.

It would be in line with  `from_dict` being a `Schema` method, not just some util function left there for advanced users.

Another rejectable solution you didn't mention would be to monkey-patch marshmallow's `Nested`...

**Comment by sirosen** [comment]
Okay, let's consider #681 "on hold". I'll put in a marshmallow PR to add the `from_dict` call to `Nested`.

Assuming nobody objects and we make that change, we'll be able to drop our copy of `Nested` in a future (major?) release of `webargs`. For users who are updating to the latest of `marshmallow` and `webargs`, they'll see no behavioral change.

Changing `marshmallow` would resolve the type-checking issue, which is what got my attention in the first place.

**Comment by sirosen** [comment]
With marshmallow-code/marshmallow#1935 merged and released, I think we can close this. I'll also ask the user on the other issue to confirm that this error still occurs. (It should not with the latest marshmallow version.)
